packages/test/test-service-load/src/loadTestDataStore.ts

```diff
@@ -91,13 +95,40 @@ class LoadTestDataStoreModel {
             throw new Error("taskmanger not available");
         }
 
+        // GC testing - We create a data store for each client pair. The url of the data store is stored in a key
+        // commong to both the clients. Each client adds a reference to this data store when it becomes a writer
+        // and removes the reference before it transtions to a reader.
+        // So, at any point in time, the data store can have 0, 1 or 2 references.
+        const halfClients = Math.floor(config.testConfig.numClients / 2);
+        const gcDataStoreUrlKey = `gc_dataStore_${config.runId % halfClients}`;
+        let gcDataStore: LoadTestDataStore | undefined;
+        if (!root.has(gcDataStoreUrlKey)) {
+            await LoadTestDataStoreModel.waitForCatchup(runtime);
+            if (!root.has(gcDataStoreUrlKey)) {
+                // The data store for this pair doesn't exist, create it and store its url.
+                gcDataStore = await LoadTestDataStoreInstantiationFactory.createInstance(containerRuntime);
+                root.set(gcDataStoreUrlKey, gcDataStore.handle.absolutePath);
+            }
```

Comment:
rather than this, can we just do  await LoadTestDataStoreModel.waitForCatchup(runtime); if !root.HasSubDir above, and then try to request the datastore, and if that fails with 404 then create it

Reply:
I am not sure what you mean. Do you mean I should move the `await LoadTestDataStoreModel.waitForCatchup(runtime)` in the beginning of this function before we create the subdirectory for this client?
Also, if the root map has the url for this data store, it should never return a 404.

Reply:
i don't want to store the absolute url. we can create the initial datastore at the same time as the subdirectory

Reply:
The problem with that is when there are 0 references to the data store, i.e., no handle, how do we get the data store to put its handle in the dds? Do we create a new one?